Storage/src/HmacKey.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Storage;
+
+use Google\Cloud\Storage\Connection\ConnectionInterface;
+
+/**
+ * Represents a Service Account HMAC key.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Storage\StorageClient;
+ *
+ * $storage = new StorageClient();
+ * $hmacKey = $storage->hmacKey($accessId);
+ * ```
+ */
+class HmacKey
+{
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * @var string
+     */
+    private $accessId;
+
+    /**
+     * @var array|null
+     */
+    private $info;
+
+    /**
+     * @var string|null
+     */
+    private $secret;
+
+    /**
+     * @param ConnectionInterface $connection A connection to Cloud Storage.
+     * @param string $projectId The current project ID.
+     * @param string $accessId The key identifier.
+     * @param array|null $info The key metadata.
+     * @param string|null $secret The key secret.
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        $projectId,
+        $accessId,
+        array $info = [],
+        $secret = null
+    ) {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->accessId = $accessId;
+        $this->info = $info;
+        $this->secret = $secret;
+    }
+
+    /**
+     * Get the HMAC Key Access ID.
+     *
+     * Example:
+     * ```
+     * $accessId = $hmacKey->accessId();
+     * ```
+     *
+     * @return string
+     */
+    public function accessId()
+    {
+        return $this->accessId;
+    }
+
+    /**
+     * Fetch the key metadata from Cloud Storage.
+     *
+     * Example:
+     * ```
+     * $keyMetadata = $hmacKey->reload();
+     * ```
+     *
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $userProject If set, this is the ID of the project which
```

Comment:
The `userProject` query string parameter is ignored for now, please mark it in the comment.

_It was omitted from the internal doc, but we clarified with the team, sorry!_

Reply:
Does this apply to all HMAC key operations, or just `getHmacKey`?